src/traces/scattergl/index.js

```diff
@@ -261,11 +261,8 @@ function sceneUpdate(gd, subplot) {
             var i;
             for(i = 0; i < scene.count; i++) {
                 if(scene.fill2d && scene.fillOptions[i]) {
-                    // must do all fills first
```

Comment:
Hmm. Looks like this broke the `gl2d_scatter_fill_self_next` mock:

![peek 2018-10-04 17-07](https://user-images.githubusercontent.com/6675409/46503279-5d8ae280-c7f8-11e8-854a-88c80ecfd989.gif)

The first two traces have `fill: 'tonext'` meaning their fills should be behind both the `data[0]` and `data[1]` markers. See corresponding SVG structure:

![image](https://user-images.githubusercontent.com/6675409/46503378-b22e5d80-c7f8-11e8-9ab9-0f209028fc0f.png)


available here -> https://rreusser.github.io/plotly-mock-viewer/#scatter_fill_self_next

Reply:
i pushed a new version .